src/Engine/EmitMSIL/Replication.cs

```diff
@@ -504,7 +679,7 @@ private static List<List<ReplicationGuide>> ConstructRepGuides(string[][] replic
 
                 foreach (List<ReplicationInstruction> replicationOption in replicationTrials)
                 {
-                    CLRFunctionEndPoint compliantTarget = GetCompliantFEP(arguments, funcGroup, replicationOption, runtimeCore, true);
+                    CLRFunctionEndPoint compliantTarget = GetCompliantFEP(thisPtr, arguments, funcGroup, replicationOption, runtimeCore, true);
```

Comment:
`GetCompliantFEP` seems to call `GetConversionDistance` - which has a TODO related to array coercion when computing type distance - I wonder if this could be related to it catching all the test cases.